aspnetcore/test/dev-tunnels.md

```diff
@@ -0,0 +1,169 @@
+
+:::image type="content" source="~/test/dev-tunnels/_static/multi-project-start.png" alt-text="Startup projects selection page showing MyWebApi and MyWebApp both starting, in that order.":::
+
+Since MyWebApi is above MyWebApp, it's started before the MyWebApp project. When the MyWebApi project is started, it will receive its tunnel URL in `VS_TUNNEL_URL`. When the MyWebApp project is started, it will receive its own tunnel URL in `VS_TUNNEL_URL` and the other project's tunnel URL in `VS_TUNNEL_URL_MyWebApi`. 
+
+To illustrate, the following highlighted lines of code have been added to the *Program.cs* file in MyWebApp:
+
+:::code language="csharp" source="~/test/dev-tunnels/sample/DevTunnelsDemo/MyWebApp/Program.cs" id="snippet1" highlight="5-8":::
+
+When the web app is started the console output looks like the following example:
+
+```console
+Tunnel URL: https://lxm0ltdt-7175.usw2.devtunnels.ms/
+API project tunnel URL: https://lxm0ltdt-7042.usw2.devtunnels.ms/
+info: Microsoft.Hosting.Lifetime[14]
+      Now listening on: https://localhost:7175
+info: Microsoft.Hosting.Lifetime[14]
+      Now listening on: http://localhost:5228
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+      Hosting environment: Development
+info: Microsoft.Hosting.Lifetime[0]
+      Content root path: C:\DevTunnelsDemo\MyWebApp
+```
+
+For information about how to set up multiple startup projects, see [How to: Set multiple startup projects](/visualstudio/ide/how-to-set-multiple-startup-projects).
+
+
+## Persistent versus temporary tunnels
+
+A persistent tunnel is one that uses the same URL after you exit and restart Visual Studio. Having a URL that doesn't change might be useful when you're integrating a web app with an external service. For example, you're implementing a GitHub webhook, or developing an API to integrate with a Power Platform app. In such cases, you need to specify the callback URL to the external service. With a persistent tunnel, you only need to configure the external service with that URL once. With a temporary tunnel, you'd have to reconfigure the external service to point to the new URL each time Visual Studio restarts.
+
+*Persistent* doesn't mean the tunnel works when Visual Studio isn't open. Any tunnel URL will connect to the local machine only if:
+
+* Visual Studio is open.
+* A tunnel is selected as active.
+* The ASP.NET Core project that the tunnel URL connects to is running.
+
+Even if the tunnel is persistent, the tunnel URL doesn't work if the associated project isn't running in Visual Studio.
+
+A temporary tunnel is fine when you only need the dev tunnel URL to work for a short time. For example, you want to share in-progress work on a web app with others, or you want to test an app on an external device. In some cases, it might be best to get a new URL each time Visual Studio starts.
```

Comment:
```suggestion
A temporary tunnel is fine when the dev tunnel URL needs to work for a short time. For example, sharing in-progress work on a web app with others, or testing an app on an external device. In some cases, it might be best to get a new URL each time Visual Studio starts.
```